modules/build/src/main/scala/scala/build/errors/BuildException.scala

```diff
@@ -3,7 +3,9 @@ package scala.build.errors
 import scala.build.Position
 
 abstract class BuildException(
-  val message: String,
-  val positions: Seq[Position] = Nil,
+  override val message: String,
+  override val positions: Seq[Position] = Nil,
   cause: Throwable = null
-) extends Exception(message, cause)
+) extends Exception(message, cause) with Diagnostic {
+  override def severity: Severity = Severity.Error
```

Comment:
We shouldn't need to put a severity here… This might lead to methods returning `Either[BuildException, T]` to return a `Left(ex)` with `ex.severity == Severity.Warning`, which doesn't make sense (exceptions on the left here are implicitly assumed to be plain errors). The severity field makes "invalid states representable".

Reply:
I've put severity here just to be able to process `BuildException` and diagnostics using the same code and hence I need to fix severity here. I can change it to `final def` to make sure nobody will treat Exception as warnings.